conda/repodata.py

```diff
@@ -0,0 +1,99 @@
+# (c) Continuum Analytics, Inc. / http://continuum.io
+# All Rights Reserved
+#
+# conda is distributed under the terms of the BSD 3-clause license.
+# Consult LICENSE.txt or http://opensource.org/licenses/BSD-3-Clause.
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+from .core import repodata as _impl
+from .connection import CondaSession
+from os.path import join
+
+
+class RepoData(object):
+    """This object represents all the package metainfo of a single channel."""
+
+    _data = {}
+
+    @staticmethod
+    def enable(url, name, priority, cache_dir=None):
+        RepoData._data[url] = RepoData(url, name, priority, cache_dir)
+
+    @staticmethod
+    def get(url):
+        return RepoData._data[url]
```

Comment:
OK, the syntax aside (we could likewise use __getitem__ in a metclass as is done in PackageData), I see your point about the KeyError. It would be more pythonic to use __getitem__ to raise the KeyError and get() to return None. So, while this slightly complicates the code, I can agree to using the same approach as PackageCache and use a metaclass such that one could use the `RepoData[url]` notation, as well as `RepoData.get(url)`. OK ?